addons/mail/models/mail_render_mixin.py

```diff
@@ -183,6 +183,32 @@ def _sub_relative2absolute(match):
 
         return html
 
+    @api.model
+    def _prepend_preview(self, body):
+        """Prepare the email body before sending.
+        Add the text preview at the beginning of the mail. The preview text is displayed
+        bellow the mail subject of most mail client (gmail, outlook...).
+        We add the character `&zwnj;` (zero-width non-joiner) to fill the end of the
+        preview in order to not have the beginning of the mail at the end of the preview
+        (don't work with simple space as the content is trimmed).
+        https://litmus.com/blog/the-ultimate-guide-to-preview-text-support
+        """
+        preview = ""
+        for element in re.finditer(r"\>([^<>]+)\<", body):
+            element = element.group(1).strip()
+            if element:
+                preview += " %s" % element
+            if len(preview) > 500:
+                break
+
+        html_preview = f"""
+            <div style="display:none;font-size:1px;height:0px;width:0px;opacity:0;">
+              {tools.html_escape(preview)} {'&zwnj;&nbsp;' * 500}
+            </div>
+        """ if preview else ''
+
+        return tools.append_content_to_html(html_preview, body, plaintext=False)
```

Comment:
Still, what if my body contains a body tag ? Maybe we should be a bit more clever about what we use, and try to find a body in body, or a main html node, in which we prepend the html preview ? No precise heuristic in mind but we should probably look at it :) .